src/HeaderChainSync/HeaderChainProvider.js

```diff
@@ -0,0 +1,189 @@
+/**
```

Comment:
The directory name should be camel case (start with a small letter) due to our naming convention.